### ISSUE TITLE
Bug: incorrect `no-undef` error for Flow Enums

### ISSUE THREAD

**Issue by mrtnzlml** [gajus/eslint-plugin-flowtype#502]
I know this sounds like it's not a problem with this plugin but bear with me.

Problematic code:

```js
// @flow strict

enum Status {
  Active,
  Paused,
  Off,
}
```

This code uses the new [Flow Enums](https://flow.org/en/docs/enums/). Eslint throws the following errors:

```text
  3:6  error  'Status' is not defined  no-undef
  4:3  error  'Active' is not defined  no-undef
  5:3  error  'Paused' is not defined  no-undef
  6:3  error  'Off' is not defined     no-undef
```

It should not throw anything (at least not `no-undef`).

We already have [`define-flow-type`](https://github.com/gajus/eslint-plugin-flowtype/tree/4265b274c7ba4e489a1e8f2dae56a25e2cb03c9e#eslint-plugin-flowtype-rules-define-flow-type) that seems to be solving the issue for other Flow specifics ([code](https://github.com/gajus/eslint-plugin-flowtype/blob/4265b274c7ba4e489a1e8f2dae56a25e2cb03c9e/src/rules/defineFlowType.js)). I _think_ we have to extend this rule to support Flow Enums correctly.

**Comment by mrtnzlml** [comment]
This issue seems to be fixed in the newly released descendant of this plugin: https://github.com/flow-typed/eslint-plugin-ft-flow
